invenio_communities/assets/semantic-ui/js/invenio_communities/community/communities-carousel/communities-carousel.js

```diff
@@ -0,0 +1,210 @@
+  }
+
+  setCarouselTimer = () => {
+    const { data } = this.state;
+    const { intervalDelay } = this.props;
+    this.setState({
+      carouselTimer: setInterval(() => {
+        data.hits.length && this.runCarousel(this.state.activeIndex + 1);
+      }, intervalDelay)
+    })
+  }
+
+  // Pause carousel when user focuses elements in the container
+  stopCarousel = () => { clearInterval(this.state.carouselTimer) }
+  startCarousel = () => { this.setCarouselTimer() }
+
+  componentWillUnmount() {
+    this.cancellableFetch && this.cancellableFetch.cancel();
+    this.stopCarousel();
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  fetchData = async () => {
+    this.setState({ isLoading: true });
+    this.cancellableFetch = withCancel(http.get(this.props.fetchUrl));
+
+    try {
+      const response = await this.cancellableFetch.promise;
+      this.setState({ data: response.data.hits, isLoading: false });
+      this.setCarouselTimer();
+
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  renderItem() {
+    const { data, activeIndex } = this.state;
+    const { defaultLogo } = this.props;
+
+    return data.hits.map((community, index) => (
+        index === activeIndex && (
+          <Item className="carousel flex align-items-center" key={community.id}>
+            <Image
+              size="small"
+              src={community.links.logo}
+              fallbackSrc={defaultLogo}
+            />
+            <Item.Content>
+              <Item.Header as={Grid} stackable className="rel-pb-1">
+                <Grid.Column computer="10" tablet="16" className="pl-0 pb-0">
+                  <Header
+                    as="a"
+                    size="medium"
+                    href={community.links.self_html}
+                  >
+                    {community.metadata.title}
+                  </Header>
+                </Grid.Column>
+                <Grid.Column computer="6" tablet="16" className="buttons pl-0 pb-0">
+                  <Button
+                    size="mini"
+                    href={community.links.self_html}
+                    content={i18next.t("Browse")}
+                  />
+                  <Button
+                    size="mini"
+                    icon="upload"
+                    labelPosition="left"
+                    positive
+                    href={`/uploads/new?community=${community.id}`}
+                    content={i18next.t("New upload")}
+                  />
+                </Grid.Column>
+              </Item.Header>
+              <Item.Description
+                content={_truncate(community.metadata.description, { length: 300 } )}
+              />
+            </Item.Content>
+          </Item>
+        )
+      )
+    )
+  }
+
+  render() {
+    const { data, animationDirection, activeIndex } = this.state;
+    const { title, animationSpeed } = this.props;
+
+    return data.hits.length ? (
```

Comment:
```suggestion
    return data.hits.length && (
```

Reply:
If we do like this, the `data.hits.length` will be printed if there are no results